File: 01x10/c-003/main.go
package main

import "fmt"

func LargestPrimeFactor(num int) int {
	for i := 2; i < num; i++ {
		if num%i == 0 {
			return LargestPrimeFactor(num / i)
		}
	}
	return num
}

func main() {
	fmt.Println(LargestPrimeFactor(600_851_475_143))
}

File: 01x10/c-003/Main.java
public class Main {

    public static long largestPrimeFactor(long num) {
        for (long i = 2; i < num; i++) {
            if (num % i == 0) {
                return largestPrimeFactor(num / i);
            }
        }
        return num;
    }

    public static void main(String[] args) {
        System.out.println(largestPrimeFactor(600_851_475_143L));
    }
}
